test/Garnet.test.cluster/ClusterManagementTests.cs

```diff
@@ -172,29 +159,171 @@ public void ClusterResetTest()
+            ClassicAssert.AreEqual(1, config.Nodes.Count);
+            ClassicAssert.AreNotEqual(nodeIds[0], node.NodeId);
+            ClassicAssert.AreEqual(0, node.Slots.Count);
+            ClassicAssert.IsFalse(node.IsReplica);
+        }
+
+        private void VerifyClusterResetFails(bool softReset = true)
+        {
+            var server = context.clusterTestUtils.GetServer(0);
+            var args = new List<object>() {
+                "reset",
+                softReset ? "soft" : "hard",
+                "60"
+            };
+
             try
             {
-                // Check DB was flushed due to hard reset
-                var resp = context.clusterTestUtils.GetServer(0).Execute("GET", "wxz");
-                ClassicAssert.IsTrue(resp.IsNull, "DB not flushed after HARD reset");
+                _ = (string)server.Execute("cluster", args);
+            }
+            catch (RedisServerException ex)
+            {
+                ClassicAssert.AreEqual("ERR CLUSTER RESET can't be called with master nodes containing keys", ex.Message);
+            }
+        }
 
-                // Add data to server
-                resp = context.clusterTestUtils.GetServer(0).Execute("SET", "wxz", "1234");
-                ClassicAssert.AreEqual("OK", (string)resp);
+        [Test, Order(4)]
+        public async Task ClusterResetDisposesGossipConnections()
+        {
+            const string meetRecv = "meet_requests_recv";
+            const string gossip = "gossip_success_count";
 
-                resp = context.clusterTestUtils.GetServer(0).Execute("GET", "wxz");
-                ClassicAssert.AreEqual("1234", (string)resp);
+            var node_count = 3;
+            context.CreateInstances(node_count, metricsSamplingFrequency: 1);
+            context.CreateConnection();
+            var endpoints = context.clusterTestUtils.GetEndpoints();
+            for (int i = 0; i < endpoints.Length - 1; i++)
+            {
+                context.clusterTestUtils.Meet(i, i + 1, context.logger);
             }
-            catch (Exception ex)
+
+            for (int i = 0; i < node_count; i++)
             {
-                context.logger?.LogError(ex, "An error occured at ClusterResetTest");
+                context.clusterTestUtils.WaitUntilNodeIsKnownByAllNodes(i);
             }
 
-            // Add node back to the cluster
-            context.clusterTestUtils.SetConfigEpoch(0, 1, context.logger);
-            context.clusterTestUtils.Meet(0, 1, context.logger);
+            context.clusterTestUtils.ClusterReset(0, soft: true);
+            await Task.Delay(1000);
 
-            context.clusterTestUtils.WaitUntilNodeIsKnownByAllNodes(0, context.logger);
+            // Let's compare and make sure there are no gossip messages coming in to the node that got reset.
+            var server = context.clusterTestUtils.GetServer(0);
+            var gossipsAfterReset = GetStat(server, "Stats", gossip);
```

Comment:
To avoid relying on the right timing below, maybe we can add field in gossipStats that returns number of open gossip connections. You can add this in ClusterProvider.cs:GetGossipStats. This way can be explicit that the connections have closed for this test. I am suprised that we didn't have this field before.